spec/index.md

```diff
@@ -431,6 +431,13 @@ This section specifies the publicly relevant paths in the tree.
 
     All partial state trees include a timestamp, indicating the time at which the state is current.
 
+### Api boundary bodes information {#state-tree-subnet}
+The state tree contains information about all API boundary nodes, which can communicate with the Internet Computer (replica nodes). The source of truth for these API boundary node records is stored in the registry canister and can be added/modified/removed via proposals.
+- `/api_boundary_nodes/<node_id>/domain` (text)
+- `/api_boundary_nodes/<node_id>/ipv4_address` (text)
```

Comment:
The `ipv4_address` is a Option<string> in Rust. Because the interface specifies it as pure text, we should not encode the `None` case (Otherwise, cbor encodes `None` as some special code). 
Instead, we just skip the fork with label `ipv4_address` in the state tree and don't populate it. Is this the expected behavior? @nikolay-komarevskiy @mraszyk

Reply:
I think that's fine and we can add a note similar to the path `/canister/<canister_id>/module_hash` to characterize that the path is sometimes omitted.

Reply:
@mraszyk add a short comment